back/spec/models/notification_spec.rb

```diff
@@ -87,49 +87,21 @@
     end
 
     describe 'project_phase_started' do
-      it 'makes project_phase_started notifications on phase started' do
-        phase = create(:active_phase)
-        project = phase.project
-        project.visible_to = 'groups'
-        project.groups << create(:group)
```

Comment:
So, you're removing all the tests related to groups, but the group logic is still in `ProjectPhaseStarted.make_notifications_on`. Why do you think we don't need to test it anymore?

Reply:
Because the latter two tests are about which moderator gets the notification. Now no moderator does (not because they are a moderator), only users who participated get the notification.

The first test is a test of groups, but implicitly this is tested by participation, since you should not be able to participate unless you have permission (in a group specified for project, for example). Which makes me wonder if we even need `participation_possible_for_context?` in the `ProjectPhaseStarted.make_notifications_on` method :thinking:

Reply:
I guess we do need it, since a user could participate in a project before permissions (groups, etc.) are added/changed and therefore not be able to participate by the time the notification is processed.

Reply:
Yes, I agree.

Reply:
Thanks, I hadn't thought of that case :+1:

Reply:
Ah I've figured it out. You don't need to use groups in your test because you don't set `project.visible_to = 'groups'`. Maybe indeed it's better not to test groups here as it's another class's responsibility. But at the same time, `visible_to: 'groups'` is not tested here `back/spec/policies/project_policy_spec.rb`. So, maybe this spec is the only place where it was checked. I mean it's checked by this line (the variable name is wrong and confusing)

```ruby
_other_admin = create(:user, email: 'user2@example.com')
```